dotnet/src/Connectors/Connectors.AzureOpenAI/Core/ClientCore.cs

```diff
@@ -1132,52 +861,51 @@ private static CompletionsOptions CreateCompletionsOptions(string text, AzureOpe
             }
         }
 
+        return options;
+    }
+
+    private static List<ChatMessage> CreateChatCompletionMessages(AzureOpenAIPromptExecutionSettings executionSettings, ChatHistory chatHistory)
+    {
+        List<ChatMessage> messages = [];
+
         if (!string.IsNullOrWhiteSpace(executionSettings.ChatSystemPrompt) && !chatHistory.Any(m => m.Role == AuthorRole.System))
         {
-            options.Messages.AddRange(GetRequestMessages(new ChatMessageContent(AuthorRole.System, executionSettings!.ChatSystemPrompt), executionSettings.ToolCallBehavior));
+            messages.AddRange(GetRequestMessages(new ChatMessageContent(AuthorRole.System, executionSettings!.ChatSystemPrompt), executionSettings.ToolCallBehavior));
```

Comment:
Do we need to create a message content to get the request?

```suggestion
            messages.AddRange(new SystemChatMessage(executionSettings!.ChatSystemPrompt));
```

Reply:
fixed, thanks!